primary/tests/integration_tests.rs

```diff
@@ -189,3 +190,217 @@ async fn test_get_collections() {
+/// by requesting them from other peers. On this example we emulate 2 authorities
+/// (2 separate primary nodes) where we store a certificate on each one. Then we
+/// are requesting via the get_collections call to the primary 1 to fetch the
+/// collections for both certificates. Since primary 1 knows only about the
+/// certificate 1 we expect to sync with primary 2 to fetch the unknown
+/// certificate 2 after it has been processed for causal completion & validation.
+///
+/// TODO: the code will not successfully fetch the batches for certificate 2
+/// since we don't block on waiting to sync the batches. That will be included
+/// in a follow up PR (see https://github.com/MystenLabs/narwhal/issues/223).
+/// Then this test will be refactored to test only for successful responses.
+#[tokio::test]
+async fn test_get_collections_with_missing_certificates() {
+    // GIVEN keys for two primary nodes
+    let mut k = keys();
+
+    let keypair_1 = k.pop().unwrap();
+    let name_1 = keypair_1.public().clone();
+
+    let keypair_2 = k.pop().unwrap();
+    let name_2 = keypair_2.public().clone();
+
+    let committee = committee();
+    let parameters = Parameters {
+        batch_size: 200, // Two transactions.
+        ..Parameters::default()
+    };
+
+    // AND create separate data stores for the 2 primaries
+    let store_primary_1 = NodeStorage::reopen(temp_dir());
+    let store_primary_2 = NodeStorage::reopen(temp_dir());
+
+    let worker_id = 0;
+
+    // AND generate and store the certificates
+    let signer_1 = keys().remove(0);
+
+    // The certificate_1 will be stored in primary 1
+    let certificate_1 = fixture_certificate(
+        signer_1,
+        store_primary_1.header_store.clone(),
+        store_primary_1.certificate_store.clone(),
+        store_primary_1.payload_store.clone(),
+        store_primary_1.batch_store.clone(),
+    )
+    .await;
+
+    // pop first to skip
+    let signer_2 = keys().remove(1);
+
+    // The certificate_2 will be stored in primary 2
+    let certificate_2 = fixture_certificate(
+        signer_2,
+        store_primary_2.header_store.clone(),
+        store_primary_2.certificate_store.clone(),
+        store_primary_2.payload_store.clone(),
+        store_primary_2.batch_store.clone(),
+    )
+    .await;
+
+    let block_ids = vec![certificate_1.digest(), certificate_2.digest()];
+
+    // Spawn the primary 1 (which will be the one that we'll interact with)
+    let (tx_new_certificates_1, _rx_new_certificates_1) = channel(CHANNEL_CAPACITY);
+    let (_tx_feedback_1, rx_feedback_1) = channel(CHANNEL_CAPACITY);
+
+    Primary::spawn(
+        name_1.clone(),
+        keypair_1,
+        committee.clone(),
+        parameters.clone(),
+        store_primary_1.header_store,
+        store_primary_1.certificate_store,
+        store_primary_1.payload_store,
+        /* tx_consensus */ tx_new_certificates_1,
+        /* rx_consensus */ rx_feedback_1,
+        /* external_consensus */ true,
```

Comment:
just an fyi when you merge with main you will have to change this too false because of [PR#231](https://github.com/MystenLabs/narwhal/pull/231)